src/core/net/srp_server.hpp

```diff
@@ -71,13 +71,22 @@ namespace Srp {
 class Server : public InstanceLocator, private NonCopyable
 {
     friend class ot::Notifier;
+    friend class UpdateMetadata;
+    friend class Service;
+    friend class Host;
```

Comment:
Make Server::AllocateId public?

Reply:
the ID is implementation-specific stuff and I think there are no other modules that will need access to it.